src/Inciter/Partitioner.h

```diff
@@ -1283,7 +1284,7 @@ class Partitioner : public CBase_Partitioner< HostProxy,
     //! Signal back to host that we are ready for partitioning the mesh
     void signal2host_setup_complete( const CProxy_Transporter& host ) {
       Group::contribute(
-        CkCallback(CkIndex_Transporter::redn_wrapper_partition(NULL), host ));
+        CkCallback(CkIndex_Transporter::redn_wrapper_part(NULL), host ));
```

Comment:
You are signaling completion with a blocking reduction. For all intents and purposes, a blocking reduction is a barrier. Hence we are using barriers to block between phases.

The alternative is to use finer grain signals and messages to indicate when the data required to proceed is available and proceed then.

Reductions should always be the last resort as that is a barrier. If you truly have no way to not use it (likely because we are still in the MPI-space) then go for it. But the general reason to use an asynchronous task based runtime/advanced programming model/whatever is to take advantage of constructs to avoid those. Unless you truly need the entire global state to be at a certain point to continue (which should only be true for when we rely on MPI libraries), figure out what data you need and proceed.

The simplest way to think of this is a stencil as that is one of the most fundamental patterns. The traditional method is to put a barrier after every time step. What should be done is to instead have the computation of a tile wait until all inputs (generally ghost cells) are provided and proceed then. This lets you avoid bulk synchronous parallelism and potentially get fun things like wavefront parallelism-esque executions. Obviously you can't always do that (maybe the data you need is a reduction on dt), but it should always be targeted until determined to not be beneficial.